### DISCUSSION TITLE
Setting value for hidden_channels in FullyConnectedNet model

### DISCUSSION THREAD

**Discussion by Kiarashdnsh** [Project-MONAI/MONAI#6749] [Q&A] [answered]
Hello,

I'm a novice in AI and image segmentation.
I have a dataset of `[256, 256, 256]`-sized images and labels. The input channel is 1, and the output channel is 3 (label IDs are 0, 1, and 2, respectively). I'm trying to segment the images using `FullyConnectedNet` with the MONAI framework. In this model's parameters, I don't know how to set the `hidden_channels`; however, I used a list of `[10]` as a placeholder since this is a required parameter.
For this model, I have used the following code:

```python
model = FCN(
        in_channels=1,
        out_channels=3,
        hidden_channels=[10],
        dropout=0.20,
)
```
and used this tutorial [here ](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/torch/unet_training_dict.py)for running the trainer.

While the trainer is running, on the `outputs = model(inputs)` line I get this error:
```python
RuntimeError: mat1 and mat2 shapes cannot be multiplied (4x884736 and 1x10)
``` 
To avoid this issue, I changed the `in_channels` to `884736` so that the two matrices could be multiplied. After this fix, on the next run, I get another error on the `loss = loss_function(outputs, labels)` line:
```python
AssertionError: ground truth has different shape (torch.Size([4, 2, 96, 96, 96])) from input (torch.Size([4, 2]))
``` 
Now I have two related questions regarding the situation:
1. How do I know what value to set in the `hidden_channels`?
2. What I've done to solve the first issue doesn't seem to be rational, since I have only 1 channel in my images. How do I solve that? I, think this issue is related to the value of `hidden_channels`. 

Thanks in advance,
Kiarash

**Comment by Kiarashdnsh** [comment]
I've read the source code for this model [here](https://docs.monai.io/en/stable/_modules/monai/networks/nets/fullyconnectednet.html#FullyConnectedNet) and still couldn't figure out how to set values for `hidden_channels`.

**Comment by KumoLiu** [comment, marked as answer]
Hi @Kiarashdnsh, thanks for your interest here.

I guess you want to use the 'fully convolutional network' (FCN) model to segment your images instead of the fully-connected network (`FullyConnectedNet`) which is used to do classification. `FullyConnectedNet` should accept one-dimensional data, shape like Batch_size*Channel. 
https://github.com/Project-MONAI/MONAI/blob/4addc5d56c26c5d00828bc8796cea73ef95372b7/monai/networks/nets/fullyconnectednet.py#L33-L35
As a first step, I recommend that you try 'UNet' (a popular variant of FCN); here is a tutorial of a basic UNet training example. (https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/torch/unet_training_array.py)

Hope it helps, thanks!

**Reply by Kiarashdnsh** [reply]
Thanks for your answer. ✔
